crates/biome_graphql_analyze/src/lint/nursery/use_unique_enum_value_names.rs

```diff
@@ -0,0 +1,82 @@
+use std::collections::HashSet;
+
+use biome_analyze::{
+    Ast, Rule, RuleDiagnostic, RuleSource, context::RuleContext, declare_lint_rule,
+};
+use biome_console::markup;
+use biome_graphql_syntax::GraphqlEnumTypeDefinition;
+use biome_rowan::AstNode;
+use biome_rule_options::use_unique_enum_value_names::UseUniqueEnumValueNamesOptions;
+use biome_string_case::StrOnlyExtension;
+
+declare_lint_rule! {
+    /// Require all enum value names to be unique.
+    ///
+    /// A GraphQL enum type is only valid if all its values are uniquely named.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```graphql,expect_diagnostic
+    /// enum A { TEST TesT }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```graphql
+    /// enum A { TEST }
+    /// ```
+    ///
+    pub UseUniqueEnumValueNames {
+        version: "next",
+        name: "useUniqueEnumValueNames",
+        language: "graphql",
+        recommended: false,
+        sources: &[RuleSource::EslintGraphql("unique-enum-value-names").same()],
+    }
+}
+
+impl Rule for UseUniqueEnumValueNames {
+    type Query = Ast<GraphqlEnumTypeDefinition>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = UseUniqueEnumValueNamesOptions;
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let mut found: HashSet<String> = HashSet::new();
```

Comment:
You should be using `TokenText` here instead of `String` if you can.

Reply:
Switched to using Cow